internal/pkg/istio/config_factory.go

```diff
@@ -63,54 +63,60 @@ func NewConfigFactory(
 // DestinationRules customize SNI in the client mTLS connection when the remote ingress is openshift-router,
 // because that ingress requires hosts compatible with https://datatracker.ietf.org/doc/html/rfc952.
 func (cf *ConfigFactory) DestinationRules() []*v1alpha3.DestinationRule {
-	remote := cf.cfg.MeshPeers.Remote
 
-	if remote.IngressType != config.OpenShiftRouter {
-		return nil
-	}
+	var destinationRules []*v1alpha3.DestinationRule
 
-	createObjectMeta := func(svcName, svcNs string) metav1.ObjectMeta {
-		return metav1.ObjectMeta{
-			Name:      fmt.Sprintf("mtls-sni-%s-%s", svcName, svcNs),
-			Namespace: cf.cfg.MeshPeers.Local.ControlPlane.Namespace,
-			Labels:    map[string]string{"federation.istio-ecosystem.io/peer": "todo"},
+	for _, remote := range cf.cfg.MeshPeers.Remotes {
+		if remote.IngressType != config.OpenShiftRouter {
+			continue
 		}
-	}
 
-	destinationRules := []*v1alpha3.DestinationRule{{
-		ObjectMeta: createObjectMeta(remote.ServiceName(), "istio-system"),
-		Spec: istionetv1alpha3.DestinationRule{
-			Host: remote.ServiceFQDN(),
-			TrafficPolicy: &istionetv1alpha3.TrafficPolicy{
-				Tls: &istionetv1alpha3.ClientTLSSettings{
-					Mode: istionetv1alpha3.ClientTLSSettings_ISTIO_MUTUAL,
-					Sni:  routerCompatibleSNI(remote.ServiceName(), "istio-system", remote.ServicePort()),
-				},
-			},
-		},
-	}}
+		createObjectMeta := func(svcName, svcNs string) metav1.ObjectMeta {
+			return metav1.ObjectMeta{
+				Name:      fmt.Sprintf("mtls-sni-%s-%s", svcName, svcNs),
```

Comment:
We must append peer name to avoid overwriting resources when 3+ meshes are connected and 2 of them export the same service (for HA) and default cluster domain is used.